services/QuillLMS/engines/evidence/lib/evidence/check.rb

```diff
@@ -4,12 +4,15 @@ module Evidence
   module Check
     class NoMatchedFeedbackTypesError < StandardError; end
 
+    USE_GEN_AI = ENV.fetch('EVIDENCE_USE_GEN_AI', 'false') == 'true'
+    AI_CHECK = USE_GEN_AI ? GenAI : AutoML
+
     ALL_CHECKS = [
       Prefilter,
       RegexSentence,
       Opinion,
       Plagiarism,
-      AutoML,
+      AI_CHECK,
       RegexPostTopic,
       Grammar,
       Spelling,
```

Comment:
Everything above is the full code change (not including bug fixes to get tests to pass).